### PR TITLE
updated power of two klee query with Nuno's suggestion

### PR DESCRIPTION
**Pull request by jubitaneja**

### CODE REVIEW

**Comment by jubitaneja** [comment]
@nunoplopes Thanks for your review comment. 1 << bitwidth is zero but I missed out the case 1 << (bitwidth-1) that gives INT_MIN. Yes, the encoding suggested by you is much simpler.

**Comment by regehr** [comment]
Good idea, thanks Nuno.  I feel bad about missing this bug in earlier code review.

**Comment by nunoplopes** [comment]
(actually my comment was bogus; this patch seems correct. I fall fool of the off-by-one mistake.  The encoding I suggested should work better, though).

**Comment by regehr** [comment]
Yes please adapt your pull req to use Nuno's suggestion Jubi.

**Comment by jubitaneja** [comment]
Updated. Thanks.

File: test/Tool/more-knownbits20.opt [hunk 1]
```diff
@@ -3,7 +3,7 @@
 ; RUN: %souper-check %solver -print-counterexample=false  %s > %t 2>&1
```
Comment (regehr) [inline test/Tool/more-knownbits20.opt]: just remove this test case, I'd say

Comment (jubitaneja) [reply test/Tool/more-knownbits20.opt]: Done.

File: test/Tool/power2.opt [hunk 2]
```diff
@@ -0,0 +1,55 @@
+; REQUIRES: solver
```
Comment (regehr) [inline test/Tool/power2.opt]: this one seems a bit brute-force, why not just assert that the ctpop == 1?

Comment (jubitaneja) [reply test/Tool/power2.opt]: I wrote this test case to make sure we have all power of two numbers (including INT_MIN) this time. ctpop anyways returned 1 when I didn't include 1 << (bitwidth-1) in my KLEE query earlier. I can add ctpop==1 test case too. Should I get rid of this long brute force test?

Comment (regehr) [reply test/Tool/power2.opt]: also this test passes with current Souper. so it doesn't test your new code.

Comment (jubitaneja) [reply test/Tool/power2.opt]: Yes, you're right. I have checked the leading zeros on the power of two now. This confirms the bug is fixed as the invalid and LGTM version of same test case just differ on no leading zeros for INT_MIN case.

**Comment by regehr** [comment]
LGTM

**Pull request merged**